client/state/sites/plans/test/trials.js

```diff
@@ -0,0 +1,311 @@
+import config from '@automattic/calypso-config';
+import {
+	PLAN_ECOMMERCE_MONTHLY,
+	PLAN_ECOMMERCE_TRIAL_MONTHLY,
+	PLAN_MIGRATION_TRIAL_MONTHLY,
+} from '@automattic/calypso-products';
+import deepFreeze from 'deep-freeze';
+import moment from 'moment';
+import { userState } from 'calypso/state/selectors/test/fixtures/user-state';
+import {
+	getECommerceTrialDaysLeft,
+	getECommerceTrialExpiration,
+	getMigrationTrialDaysLeft,
+	getMigrationTrialExpiration,
+	isECommerceTrialExpired,
+	isMigrationTrialExpired,
+	isSiteOnECommerceTrial,
+	isSiteOnMigrationTrial,
+} from '../selectors';
+
+// Mock the config system
+jest.mock( '@automattic/calypso-config', () => {
+	const mock = () => '';
+	mock.isEnabled = jest.fn( () => true );
+
+	return mock;
+} );
+
+// Mock the Migration Trial feature flag (plans/migration-trial)
+const withTrialEnabled = () =>
+	config.isEnabled.mockImplementation( ( key ) => key === 'plans/migration-trial' );
+const withTrialDisabled = () =>
+	config.isEnabled.mockImplementation( ( key ) => key !== 'plans/migration-trial' );
+
+describe( 'trials', () => {
+	beforeEach( () => {
+		withTrialEnabled();
+	} );
+
+	afterEach( () => {
+		jest.resetAllMocks();
+	} );
+
+	function getState( plan, siteId ) {
+		return deepFreeze( {
+			...userState,
+			sites: {
+				plans: {
+					[ siteId ]: plan ? { data: [ plan ] } : { data: [] },
+				},
+				items: {
+					[ siteId ]: {
+						URL: 'https://example.wordpress.com',
+					},
+				},
+			},
+			siteSettings: {
+				items: {},
+			},
+		} );
+	}
+
+	describe( '#isSiteOnECommerceTrial()', () => {
+		const siteId = 1337;
+		test( 'Should return true when the e-commerce trial is in the purchases list', () => {
+			const plan = {
+				ID: 1,
+				productSlug: PLAN_ECOMMERCE_TRIAL_MONTHLY,
+				blogId: siteId,
+				currentPlan: true,
+			};
+
+			const state = getState( plan, siteId );
```

Comment:
Small nitpick here, this plan setup seems to be appearing multiple times in the file, maybe we can turn this into a helper function  by passing `siteId` and `productSlug` into it?